src/PKSim.Presentation/Presenters/Snapshots/LoadProjectFromSnapshotPresenter.cs

```diff
@@ -31,7 +45,27 @@ public PKSimProject LoadProject()
       protected override async Task<IEnumerable<PKSimProject>> LoadModelAsync(string snapshotFile)
       {
          var project = await _snapshotTask.LoadProjectFromSnapshot(snapshotFile);
-         return new[] {project};
+         _registrationTask.RegisterProject(project);
+         await runQualificationPlans(project);
+         return new[] { project };
+      }
+
+      protected override void ClearModel(IEnumerable<PKSimProject> model)
+      {
+         var projects = model?.ToList();
+         base.ClearModel(projects);
+         _registrationTask.UnregisterProject(projectFrom(projects));
+      }
+
+      private PKSimProject projectFrom(IEnumerable<PKSimProject> projects) => projects?.FirstOrDefault();
+
+      private async Task runQualificationPlans(PKSimProject project)
```

Comment:
This is the runner that is done once the snapshot has been loaded. Much cleaner that way